### PR TITLE
Fix for idna tr domains

### PR DESCRIPTION
**Pull request by ofaruk89**
.tr domains not need Punycode conversion, if you convert them they will not be able to get whois information.

**CODE**
`domain_info = whois.whois("s-vasıta.com.tr")
        print(domain_info)

`
**OUTPUT BEFORE FIX**

`{
  "domain_name": null,
  "creation_date": null,
  "expiration_date": null,
  "frozen_status": null,
  "status": null,
  "name_servers": null,
  "registrant_name": null,
  "admin": null,
  "admin_organization": null,
  "admin_address": null,
  "admin_phone": null,
  "admin_fax": null,
  "tech": null,
  "tech_organization": null,
  "tech_address": null,
  "tech_phone": null,
  "tech_fax": null,
  "billing": null,
  "billing_organization": null,
  "billing_address": null,
  "billing_phone": null,
  "billing_fax": null
}`
**AFTER FIX**

`{
  "domain_name": "s-vasıta.com.tr",
  "creation_date": "2022-09-14 00:00:00",
  "expiration_date": "2025-09-13 00:00:00",
  "frozen_status": "-",
  "status": "The domain is LOCKED to transfer.",
  "name_servers": "ns1.ihsdnsx8.com\nns2.ihsdnsx8.com",
  "registrant_name": "Sahibinden Bilgi Teknolojileri Pazarlama ve Ticaret A.Ş.\n   Hidden upon user request\n   Hidden upon user request\n   Hidden upon user request",
  "admin": null,
  "admin_organization": null,
  "admin_address": null,
  "admin_phone": null,
  "admin_fax": null,
  "tech": null,
  "tech_organization": null,
  "tech_address": null,
  "tech_phone": null,
  "tech_fax": null,
  "billing": null,
  "billing_organization": null,
  "billing_address": null,
  "billing_phone": null,
  "billing_fax": null
}`

### CODE REVIEW

File: whois/whois.py [hunk 1]
```diff
@@ -255,6 +255,8 @@ def choose_server(self, domain):
         """Choose initial lookup NIC host"""
         try:
             domain = domain.encode("idna").decode("utf-8")
+            if domain.endswith(".tr"):
+                domain = domain
```
Comment (richardpenman) [inline whois/whois.py:259]: what is this supposed to do?

**Comment by richardpenman** [comment]
There might be other TLDs that want to disable punycode conversion. Instead of hardcoding this for TR could you add a `convert_punycode` flag with the default set to True?

**Comment by ofaruk89** [comment]
> There might be other TLDs that want to disable punycode conversion. Instead of hardcoding this for TR could you add a `convert_punycode` flag with the default set to True?

okay i will set, give me a few minute

**Comment by ofaruk89** [comment]
**Its okay**

**Comment by richardpenman** [comment]
nice work, thanks for that! 

Apparently this breaks some of the test cases due to text / bytes encoding:
https://github.com/richardpenman/whois/actions/runs/11615445257/job/32367304331?pr=245
Would you be able to look at that? You can run tests locally with `python -m pytest`.

**Comment by ofaruk89** [comment]
> nice work, thanks for that!
> 
> Apparently this breaks some of the test cases due to text / bytes encoding: https://github.com/richardpenman/whois/actions/runs/11615445257/job/32367304331?pr=245 Would you be able to look at that? You can run tests locally with `python -m pytest`.

fix is ready

**Pull request merged**